core/src/main/java/org/jruby/Main.java

```diff
@@ -531,6 +537,26 @@ protected static int handleRaiseException(RaiseException rj) {
         }
     }
 
+    public static void printTruffleTimeMetric(String id) {
+        if (Options.TRUFFLE_METRICS_TIME.load()) {
+            final long millis = System.currentTimeMillis();
+            System.err.printf("%s %d.%03d%n", id, millis / 1000, millis % 1000);
+        }
+    }
+
+    private static void printTruffleMemoryMetric() {
+        if (Options.TRUFFLE_METRICS_MEMORY_USED_ON_EXIT.load()) {
+            System.gc();
```

Comment:
This seems to be problematic for multiple reasons.

Haven't yet read further in the changes, but, you are in the main method, and do a GC. So, I would expect, if there is no static state/memory leaks, after this GC, the used memory should be 0.
So, you might want to get the heap usage before the GC.

Also, System.gc() is really just a hint on HotSpot, you have to call it 3-4 times in succession to motivate HotSpot to actually do a collect (but, this then usually works).